pkg/kubectl/cmd/util/helpers.go

```diff
@@ -439,6 +440,9 @@ func Merge(codec runtime.Codec, dst runtime.Object, fragment, kind string) (runt
 	if err != nil {
 		return nil, err
 	}
+	if err = resource.ValidateSchema(patched, schema); err != nil {
+		return nil, fmt.Errorf("error validating resource json after merging with override string: %s\n%v", patched[:], err)
```

Comment:
this sounds misleading: the printed string is the result of merging, not the override string.